src/braket/simulator_v2/julia_workers.py

```diff
@@ -0,0 +1,84 @@
+import json
+import sys
+from collections.abc import Sequence
+from typing import List, Optional, Union
+
+from braket.ir.openqasm import Program as OpenQASMProgram
+
+
+def _handle_julia_error(error):
+    import sys
+
+    if isinstance(error, sys.modules["juliacall"].JuliaError):
+        python_exception = getattr(error.exception, "alternate_type", None)
+        if python_exception is None:
+            # convert to RuntimeError as JuliaError can't be serialized
+            py_error = RuntimeError(
+                "Unable to unwrap internal Julia exception."
+                f"Exception message: {str(error.exception.message)}"
+            )
+        else:
+            class_val = getattr(sys.modules["builtins"], str(python_exception))
+            py_error = class_val(str(error.exception.message))
+        raise py_error
+    else:
+        raise error
+    return
+
+
+def translate_and_run(
+    device_id: str, openqasm_ir: OpenQASMProgram, shots: int = 0
+) -> str:
+    jl = sys.modules["juliacall"].Main
+    jl_shots = shots
+    jl_inputs = json.dumps(openqasm_ir.inputs) if openqasm_ir.inputs else "{}"
+    try:
+        result = jl.BraketSimulator.simulate(
+            device_id,
+            openqasm_ir.source,
+            jl_inputs,
+            jl_shots,
+        )
+
+    except Exception as e:
+        _handle_julia_error(e)
+
+    return result
+
+
+def translate_and_run_multiple(
+    device_id: str,
+    programs: Sequence[OpenQASMProgram],
+    shots: Optional[int] = 0,
+    inputs: Optional[Union[dict, Sequence[dict]]] = {},
+) -> List[str]:
+    jl = sys.modules["juliacall"].Main
+    irs = [program.source for program in programs]
+    is_single_input = isinstance(inputs, dict) or len(inputs) == 1
+    py_inputs = {}
+    if (is_single_input and isinstance(inputs, dict)) or not is_single_input:
+        py_inputs = [inputs.copy() for _ in range(len(programs))]
+    elif is_single_input and not isinstance(inputs, dict):
+        py_inputs = [inputs[0].copy() for _ in range(len(programs))]
+    else:
+        py_inputs = inputs
```

Comment:
Please double-check my logic, but I think this is equivalent and makes the logic more readable:
```suggestion
    py_inputs = {}
    if len(inputs) > 1 or isinstance(inputs, dict)):
        py_inputs = [inputs.copy() for _ in range(len(programs))]
    elif len(inputs) == 1:
        py_inputs = [inputs[0].copy() for _ in range(len(programs))]
```